Kwc/Abstract.php

```diff
@@ -411,15 +424,52 @@ public function getMasterTemplateVars(Kwf_Component_Data $innerComponent, Kwf_Co
             usort($ret['multiBoxes'][$box], array('Kwf_Component_View_Helper_ComponentWithMaster', '_sortByPriority'));
         }
 
-        $ret['cssClass'] = 'frontend';
-        $cls = Kwc_Abstract::getSetting($this->getData()->componentClass, 'processedCssClass');
-        foreach (explode(' ', $cls) as $i) {
-            $i = 'kwfup-master'.ucfirst($i);
-            $ret['cssClass'] .= ' '.$i;
+        $ret['cssClass'] = 'kwfup-frontend';
+
+        if (Kwc_Abstract::hasSetting($this->getData()->componentClass, 'masterTemplate')) {
+            $masterTemplate = Kwc_Abstract::getSetting($this->getData()->componentClass, 'masterTemplate');
+        } else {
+            $masterTemplate = $renderer->getTemplate($this->getData(), 'Master');
```

Comment:
Not directly connected with this pull request, but to avoid code duplication: move to $renderer->getTemplate(), which does an if($type == 'Master')